Attempt to do a comparison of forced UseVirtualTag

manofstick opened the pull request:
Just creating PR to get AppVeyor to do some work...

This is just a shotgun approach to UseVirtualTag, and still tweaks are required, but should be a reasonable baseline...

manofstick committed:
Attempt to do a comparison of forced UseVirtualTag

forki commented on build.fsx:
```diff
@@ -10,8 +10,8 @@ open System.IO
-let compilerForkToTest = @"https://github.com/forki/visualfsharp"
-let compilerHashes = ["71c8798e19d6e15d3e6a98c80da658aa5ed2c630"; "ee0baed"]
+let compilerForkToTest = @"https://github.com/manofstick/visualfsharp"
+let compilerHashes = ["5c931e3149ebdd5d1e5f9721a3c6dbfb63e9d1c6"; "9d5e36a7aa66f029640aa05f96bd7b54907aef81"]
```
Maybe we need some way to alias. I assume branch names will just work

forki commented:
```
Compiler: 5c931e3149ebdd5d1e5f9721a3c6dbfb63e9d1c6
   Project: Paket 11872.0ms
   Project: FSharpx.Collections 2524.0ms
   Project: FSharp.Data 4001.0ms
   Project: SQLProvider 7960.0ms
   Average Run: 26358.0ms
Compiler: 9d5e36a7aa66f029640aa05f96bd7b54907aef81
   Project: Paket 11486.0ms
   Project: FSharpx.Collections 2404.0ms
   Project: FSharp.Data 3874.0ms
   Project: SQLProvider 7598.0ms
   Average Run: 25363.0ms
```

forki commented:
this looks awesome. Only question is how trustworthy my perf framework is...

manofstick committed:
Updated hash

manofstick committed:
Trigger a build

manofstick commented:
Well a bit more what I was expecting:

Compiler: 32c530ebe42f34a863a9e55b682f90685e658125
   Project: Paket 11217.0ms
   Project: FSharpx.Collections 2424.0ms
   Project: FSharp.Data 3801.0ms
   Project: SQLProvider 7483.0ms
   Average Run: 24926.0ms
Compiler: 9d5e36a7aa66f029640aa05f96bd7b54907aef81
   Project: Paket 11266.0ms
   Project: FSharpx.Collections 2417.0ms
   Project: FSharp.Data 3831.0ms
   Project: SQLProvider 7383.0ms
   Average Run: 24898.0ms

But times obviously jump around a bit; because the unmodified changeset went from:

Compiler: 9d5e36a7aa66f029640aa05f96bd7b54907aef81
   Average Run: 25363.0ms (run from other day)
to
   Average Run: 24898.0ms (today's run)

Which is an almost 2% change; so probably hard to detect any minor changes, but atleast should detect any major problems...

manofstick committed:
Check when Map and Set use UseVirtualTag

manofstick committed:
Checking alternative List.filter

Shouldn't really change the performance at all, so just check that that
is true, within the scope of the AppVeyor environment...

manofstick committed:
Flipping the build ordering

See if the results are consistent...
